src/main/resources/db/migration/V1_132__print_files_to_clob_update.sql

```diff
@@ -0,0 +1,216 @@
+DROP PROCEDURE juror_mod.printfiles_to_clob(int4);
+
+CREATE OR REPLACE PROCEDURE juror_mod.printfiles_to_clob(IN p_document_limit integer DEFAULT 1000000)
+ LANGUAGE plpgsql
+AS $procedure$
+DECLARE
+	v_form_type VARCHAR(6);
+	v_max_rec_len INTEGER;
+	v_ext_date DATE;
+	forms RECORD;
+	v_count INTEGER:=0;
+
+BEGIN
+	v_ext_date :=	CASE
+						WHEN TO_CHAR(NOW(),'SSSSS')::INTEGER < 64800
+              				THEN NOW()::date - 1
+               				ELSE NOW()
+              		END;
+
+    -- delete redundant letter requests
+	CALL juror_mod.delete_printfiles();
+
+	FOR forms IN
+		SELECT	fr.form_type,
+				fr.max_rec_len::integer
+		FROM juror_mod.t_form_attr fr
+
+		loop
+			CALL juror_mod.write_printfiles(forms.form_type, forms.max_rec_len, v_ext_date, p_document_limit);
+		END LOOP;
+
+END;
+$procedure$
+;
+
+DROP PROCEDURE juror_mod.write_printfiles(in varchar, in int4, in date, in int4, inout int4);
+
+CREATE OR REPLACE PROCEDURE juror_mod.write_printfiles(IN p_form_type character varying, IN p_max_rec_len integer, IN p_ext_date date, IN p_document_limit integer)
+ LANGUAGE plpgsql
+AS $procedure$
+DECLARE
+	form_details RECORD; -- current record from bulk print data table.
+	v_filename VARCHAR(50); -- current filename
+	v_header VARCHAR(50); -- header for print file
+	v_row_id INTEGER; -- row id of current record from bulk print data
+	v_data text:=''; -- data to be written to file
+	v_detail_rec text; -- row of data from bulk print data
+	v_revision int8; -- sequence number for file name
+	v_page_size integer; -- total pages for form type
+	v_total_documents_size integer; -- total amount of remaining records to write
+	v_documents_to_write integer; -- total amount of documents to write to current file
+
+begin
+	select count(bpd.id)
+	from juror_mod.bulk_print_data bpd
+	WHERE bpd.form_type = p_form_type
+	AND COALESCE(bpd.extracted_flag,'N') = 'N'
```

Comment:
Will the COALESCE adversely affect the use of any database indexing that we are relying on to efficeintly find the rows to be extracted? In the old Oracle database, we had to have a function based index in order for the index to be used.

Do we know whether EXTRACTED_FLAG could be null? We won't need the COALESCE if we are sure it is always Y or N.

(converting EXTRACTED_FLAG to a boolean during data migration might have been better with hindsight).

Reply:
Updated to use false instead of 'N'